client/landing/stepper/declarative-flow/internals/steps-repository/woo-transfer/index.tsx

```diff
@@ -1,20 +1,152 @@
+				action: failureInfo.type,
+				site: site?.URL,
+				code: failureInfo.code,
+			},
+		} );
+	};
 
 	useEffect( () => {
+		if ( ! siteId ) return;
+
 		setPendingAction( async () => {
-			setProgressTitle( 'Transfering site' );
-			await wait( 3000 );
-			// TODO actually transfer
+			const startTime = new Date().getTime();
+			const totalTimeout = 1000 * 180;
+			const maxFinishTime = startTime + totalTimeout;
+
+			// Initiate transfer
+			await initiateAtomicTransfer( siteId, 'woo-on-plans' );
+
+			// Poll for transfer status
+			let stopPollingTransfer = false;
+
+			while ( ! stopPollingTransfer ) {
+				await wait( 3000 );
+				await requestLatestAtomicTransfer( siteId );
+				const transfer = getSiteLatestAtomicTransfer( siteId );
+				const transferError = getSiteLatestAtomicTransferError( siteId );
+				const transferStatus = transfer?.status;
+				const isTransferringStatusFailed = transferError && transferError?.status >= 500;
+
+				switch ( transferStatus ) {
+					case transferStates.PENDING:
+						setProgress( 0.2 );
+						break;
+					case transferStates.ACTIVE:
+						setProgress( 0.4 );
+						break;
+					case transferStates.PROVISIONED:
+						setProgress( 0.5 );
+						break;
+					case transferStates.COMPLETED:
+						setProgress( 0.7 );
+						break;
+				}
+
+				if ( isTransferringStatusFailed || transferStatus === transferStates.ERROR ) {
+					handleTransferFailure( {
+						type: 'transfer',
+						error: transferError?.message || '',
+						code: transferError?.code || '',
+					} );
+					throw new Error( 'transfer error' );
+				}
+
+				if ( maxFinishTime < new Date().getTime() ) {
+					handleTransferFailure( {
+						type: 'transfer_timeout',
+						error: 'transfer took too long',
+						code: 'transfer_timeout',
+					} );
+					throw new Error( 'transfer timeout' );
+				}
+
+				stopPollingTransfer = transferStatus === transferStates.COMPLETED;
+			}
+
+			// Poll for software status
+			let stopPollingSoftware = false;
+
+			while ( ! stopPollingSoftware ) {
+				await requestAtomicSoftwareStatus( siteId, 'woo-on-plans' );
+				const softwareStatus = getAtomicSoftwareStatus( siteId, 'woo-on-plans' );
+
+				const softwareError = getAtomicSoftwareError( siteId, 'woo-on-plans' );
+				if ( softwareError ) {
+					handleTransferFailure( {
+						type: 'transfer',
+						error: softwareError.message || '',
+						code: softwareError.status || '',
+					} );
+					throw new Error( 'software install error' );
+				}
+
+				if ( maxFinishTime < new Date().getTime() ) {
+					handleTransferFailure( {
+						type: 'transfer_timeout',
+						error: 'transfer took too long',
+						code: 'transfer_timeout',
+					} );
+					throw new Error( 'transfer timeout' );
+				}
+
+				stopPollingSoftware = !! softwareStatus?.applied;
+
+				if ( ! stopPollingSoftware ) await wait( 3000 );
+			}
+			setProgress( 1 );
+			// Allow progress bar to complete
+			await wait( 500 );
```

Comment:
This is something weird, I don't think we should be stalling for some visual effect right?

Reply:
I agree with you actually! I just kept this logic from the old/current version so the behavior is the same.

Reply:
Ok, let's remove it then! Making the user wait without real reason feels bad